acvm/src/pwg/mod.rs

```diff
@@ -179,16 +190,19 @@ impl<B: BlackBoxFunctionSolver> ACVM<B> {
     ///
     /// The ACVM can then be restarted to solve the remaining Brillig VM process as well as the remaining ACIR opcodes.
     pub fn resolve_pending_foreign_call(&mut self, foreign_call_result: ForeignCallResult) {
+        if !matches!(self.status, ACVMStatus::RequiresForeignCall(_)) {
+            panic!("ACVM is not expecting a foreign call response as no call was made");
+        }
```

Comment:
This addresses the request to print the opcode when panicking due to it not being a brillig opcode.

This is the actual sitation we want to check for and `instruction_pointer` pointing at a brillig opcode is a consequence of this. I've then made a non-brillig opcode an `unreachable` panic.